packages/realm-web/src/Credentials/Credentials.ts

```diff
@@ -0,0 +1,30 @@
+import type { AnonymousCredentials } from "./AnonymousCredentials";
+import type { UsernamePasswordCredentials } from "./UsernamePasswordCredentials";
+
+/**
+ * Abstract base class for credentials.
+ * Exposes concrete types of credentials as static members and methods.
+ */
+export abstract class Credentials {
+    // tslint:disable:completed-docs
+    static AnonymousCredentials: typeof AnonymousCredentials;
+    static UsernamePasswordCredentials: typeof UsernamePasswordCredentials;
+    // tslint:endable:completed-docs
+
+    /**
+     * The name of the authentication provider used when authenticating
+     */
+    public readonly providerName: string;
+
+    constructor(providerName: string) {
+        this.providerName = providerName;
+    }
+
+    static anonymous(providerName?: string) {
```

Comment:
It's a pattern we have in [our current Realm JS API](https://realm.io/docs/javascript/latest/api/Realm.Sync.Credentials.html).

It's meant as shorthand to help people write less code.

```typescript
import Realm from "realm";
const credentials = Realm.Credentials.anonymous();
```

instead of

```typescript
import Realm from "realm";
const credentials = new Realm.Credentials.AnonymousCredentials();
```

Admittedly, people could simply write (but thats less discoverable through intellisense).

```typescript
import { AnonymousCredentials } from "realm";
const credentials = new AnonymousCredentials();
```

Reply:
This was discussed in a Zoom meeting. I've pushed an update with the agreed changes, please re-review that part :)